### ISSUE TITLE
Image as inline attachments not being showed

### ISSUE THREAD

**Issue by pablocrivella** [markets/maily#30]
### Description

When loading images from an inline attachment url on the email, they don't show correctly:

![maily - account mailer deletion warning 2018-09-18 11-01-55](https://user-images.githubusercontent.com/1854923/45676610-6ca03e00-bb32-11e8-8351-75f42d701685.png)

**Comment by markets** [comment]
Hi @pablocrivella, are you using inline attachments right? Because in all my current Maily installations I use regular sprockets helpers:

    <%= image_tag 'notifier/crisalix-logo.png', align: 'left' %>

And it works with no problems, example:

![captura de pantalla 2018-09-18 a la s 15 37 46](https://user-images.githubusercontent.com/576701/45691415-e6e3b900-bb58-11e8-8831-352305e7dfdc.png)

If that's the case (inline attachments), what src/url are you seeing when inspecting those images?

**Comment by pablocrivella** [comment]
Yes, inline attachments i what i meant :)

```
src="cid:5ba11490949d3_5b793fcc8515c81823670@mylaptop.local.mail"
```

**Comment by markets** [comment]
Ok @pablocrivella I just made a quick look using inline attachments and now I see this `cid:` urls, but to be honest, I have no idea how to fix this in a non-hacky way, for now. I found a temporal workaround here (https://stackoverflow.com/questions/26574811/rails-4-mailer-preview-preview-attached-images). Basically it changes content to embedded `base64` data:

```ruby
# put this into an initializer
require 'mail'

module InlineMailPartUrls
  def url
    "data:#{mime_type};base64,#{Base64.encode64(body.decoded)}"
  end
end

class Mail::Part
  prepend InlineMailPartUrls
end
```

**Comment by markets** [comment]
You need to patch it on each env you are running Maily and want to properly show inline images, but take into account that for real emails, this will also work the same way (base 64).

Another option, is to use directly images from `app/assets/images/*` + `config.action_mailer.asset_host = 'https://your-hostname.com'` and avoid inline attachments. This is what I usually do, I remember I have had some issues with inline images in mobile clients in the past.

That said, let's keep this open for some time, it would be nice to find a more appropriate fix or workaround for these cases.

**Comment by pablocrivella** [comment]
Yeah i also normally go for the `config.action_mailer.asset_host` option but in this case we can't/shouldn't access the host due to security reasons :) 

Anyway, thanks a lot for all your feedback!

Feel free to close this issue if you want to

**Comment by dmitry** [comment]
Would be nice to have a support for inline attachments. That's how rails_email_preview provided support for that: https://github.com/glebm/rails_email_preview/commit/8a6b13606a7780c2c43678cd2c378f1009f91f73

**Comment by dmitry** [comment]
And that's how rails previewer did that: https://github.com/rails/rails/blob/master/actionmailer/lib/action_mailer/inline_preview_interceptor.rb

Seems like better approach, as it doesn't require to fetch files via `router->controller` path. Simply replace `cid` with `base64` image contents.

**Comment by markets** [comment]
Totally agree @dmitry! The changes commented here https://github.com/markets/maily/issues/30#issuecomment-422958523 are really hacky :)

I'd like to add real support for inline attachments. The path taken by Rails seems simpler 👌, if someone would pick it up and send a PR based on that I'll very really happy to review and merge 🚀

**Comment by markets** [comment]
@pablocrivella @dmitry Inline attachments support released in v0.12.0 (https://rubygems.org/gems/maily/versions/0.12.0) 🚀
